lib/ancestry/materialized_path.rb

```diff
@@ -50,6 +50,14 @@ def sibling_conditions(object)
       t[ancestry_column].eq(node[ancestry_column])
     end
 
+    # idintifies leaves that belongs to the object (excluding itself)
+    def leaf_conditions(object)
+      t = arel_table
+      node = to_node(object)
+      all_ancestor_ids = self.ancestry_base_class.all.map(&:ancestor_ids).flatten.uniq
```

Comment:
Is it possible to determine these ids in a query and not bring back all the records?
This is the main reason why I've been reluctant to introduce `leafs` method. I don't want to bring back every record.
This brings back every record, determines ids, and then brings back more records.
Would have been better to keep `all` in a local variable and delete from there.

But again, I want to find a sql way to determine the leafs.

aside 2: Logic like this is the driver for my proposed MaterializedPathArray changes.

Reply:
As you say this approach can be very costly - perhaps way too costly just to obtain a few number of records.
Let me come back with a new PR (if you are okay) as I already have pure sql based approaches at hand. Would be nice if I can better understand how you see it / whether or not it makes sense.